test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -0,0 +1,258 @@
+                  .output("/dev/null")
+                  .block_size(block_size)
+                  .count(blocks_300MiB))
+            dd.run()
+            os_utils.sync()
+            os_utils.drop_caches()
+            dd = (Dd()
+                  .input(core2.system_path)
+                  .output("/dev/null")
+                  .block_size(block_size)
+                  .count(blocks_300MiB))
+            dd.run()
+        except Exception:
+            TestRun.LOGGER.error("The remaining core is not able to use cache.")
+
+    with TestRun.step("Stop cache."):
+        casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cache_mode", [CacheMode.WO, CacheMode.WT, CacheMode.WB, CacheMode.WA])
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_one_core_release(cache_mode):
+    """
+        title: Test if OpenCAS dynamically allocates space according to core devices needs.
+        description: |
+          When one or more core devices are unused in a single cache instance all blocks
+          previously occupied should be available to other core devices.
+          Test is without pass through mode.
+        pass_criteria:
+          - No system crash.
+          - The remaining core is able to use cache.
+          - OpenCAS frees blocks occupied by unused core and allocates it to the remaining core.
+    """
+    with TestRun.step("Prepare two cache and one core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(512, Unit.MebiByte)])
+        cache_part = cache_dev.partitions[0]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(1, Unit.GibiByte)] * 2)
+        core_part1 = core_dev.partitions[0]
+        core_part2 = core_dev.partitions[1]
+
+    with TestRun.step("Start cache"):
+        cache = casadm.start_cache(cache_part, cache_mode, force=True)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Add both core devices to cache."):
+        core1 = cache.add_core(core_part1)
+        core2 = cache.add_core(core_part2)
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 2:
+            TestRun.fail(f"Expected cores count: 2; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Change sequential cutoff policy to 'never'."):
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Fill cache with pages from the first core."):
+        blocks = int(cache.size.value / block_size.value)
+        if cache_mode != CacheMode.WO:
+            dd = (Dd()
+                  .input(core1.system_path)
+                  .output("/dev/null")
+                  .block_size(block_size)
+                  .count(blocks))
```

Comment:
Actually you read the whole cache (512MiB), not only 100MiB - didn't you want to read only 100MiB?

Reply:
For the first time - no, because I want to fill cache with data from the first core and then check, if usage of the second core would overwrite primarily cached data.